testlapack/dgeev.go

```diff
@@ -0,0 +1,693 @@
+			evWant: Grcar{N: 50, K: 30}.Eigenvalues(),
+		},
+		{
+			a:      Grcar{N: 150, K: 2}.Matrix(),
+			evWant: Grcar{N: 150, K: 2}.Eigenvalues(),
+		},
+		{
+			a:      Grcar{N: 150, K: 148}.Matrix(),
+			evWant: Grcar{N: 150, K: 148}.Eigenvalues(),
+		},
+
+		{
+			a:      Hanowa{N: 6, Alpha: 17}.Matrix(),
+			evWant: Hanowa{N: 6, Alpha: 17}.Eigenvalues(),
+		},
+		{
+			a:      Hanowa{N: 50, Alpha: -1}.Matrix(),
+			evWant: Hanowa{N: 50, Alpha: -1}.Eigenvalues(),
+		},
+		{
+			a:      Hanowa{N: 100, Alpha: -1}.Matrix(),
+			evWant: Hanowa{N: 100, Alpha: -1}.Eigenvalues(),
+		},
+
+		{
+			a:      Lesp(2).Matrix(),
+			evWant: Lesp(2).Eigenvalues(),
+		},
+		{
+			a:      Lesp(3).Matrix(),
+			evWant: Lesp(3).Eigenvalues(),
+		},
+		{
+			a:      Lesp(4).Matrix(),
+			evWant: Lesp(4).Eigenvalues(),
+		},
+		{
+			a:      Lesp(5).Matrix(),
+			evWant: Lesp(5).Eigenvalues(),
+		},
+		{
+			a:      Lesp(10).Matrix(),
+			evWant: Lesp(10).Eigenvalues(),
+		},
+		{
+			a:      Lesp(15).Matrix(),
+			evWant: Lesp(15).Eigenvalues(),
+		},
+		{
+			a:      Lesp(30).Matrix(),
+			evWant: Lesp(30).Eigenvalues(),
+		},
+		{
+			a:      Lesp(50).Matrix(),
+			evWant: Lesp(50).Eigenvalues(),
+			tol:    1e-12,
+		},
+		{
+			a:      Lesp(101).Matrix(),
+			evWant: Lesp(101).Eigenvalues(),
+			tol:    1e-12,
+		},
+		{
+			a:      Lesp(150).Matrix(),
+			evWant: Lesp(150).Eigenvalues(),
+			tol:    1e-12,
+		},
+
+		{
+			a:      Rutis{}.Matrix(),
+			evWant: Rutis{}.Eigenvalues(),
+		},
+
+		{
+			a:      Tris{N: 74, X: 1, Y: -2, Z: 1}.Matrix(),
+			evWant: Tris{N: 74, X: 1, Y: -2, Z: 1}.Eigenvalues(),
+		},
+		{
+			a:      Tris{N: 74, X: 1, Y: 2, Z: -3}.Matrix(),
+			evWant: Tris{N: 74, X: 1, Y: 2, Z: -3}.Eigenvalues(),
+		},
+		{
+			a:      Tris{N: 75, X: 1, Y: 2, Z: -3}.Matrix(),
+			evWant: Tris{N: 75, X: 1, Y: 2, Z: -3}.Eigenvalues(),
+		},
+
+		{
+			a:      Wilk4{}.Matrix(),
+			evWant: Wilk4{}.Eigenvalues(),
+		},
+		{
+			a:      Wilk12{}.Matrix(),
+			evWant: Wilk12{}.Eigenvalues(),
+			tol:    1e-8,
+		},
+		{
+			a:      Wilk20(0).Matrix(),
+			evWant: Wilk20(0).Eigenvalues(),
+		},
+		{
+			a:      Wilk20(1e-10).Matrix(),
+			evWant: Wilk20(1e-10).Eigenvalues(),
+			tol:    1e-12,
+		},
+
+		{
+			a:      Zero(1).Matrix(),
+			evWant: Zero(1).Eigenvalues(),
+		},
+		{
+			a:      Zero(10).Matrix(),
+			evWant: Zero(10).Eigenvalues(),
+		},
+		{
+			a:      Zero(50).Matrix(),
+			evWant: Zero(50).Eigenvalues(),
+		},
+		{
+			a:      Zero(100).Matrix(),
+			evWant: Zero(100).Eigenvalues(),
+		},
+	} {
+		for _, jobvl := range []lapack.JobLeftEV{lapack.ComputeLeftEV, lapack.None} {
+			for _, jobvr := range []lapack.JobRightEV{lapack.ComputeRightEV, lapack.None} {
+				for _, extra := range []int{0, 11} {
+					for _, optwork := range []bool{false, true} {
+						testDgeev(t, impl, strconv.Itoa(i), test, jobvl, jobvr, extra, optwork)
+					}
+				}
+			}
+		}
+	}
+
+	// for _, n := range []int{10, 30, 50} {
```

Comment:
Why don't you generate random eigenvalues and eigenvectors, multiply the matrices, and then run the algorithm?